hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/Route.java

```diff
@@ -188,29 +210,40 @@ public void configure() {
 				     .log("Received response from RTrans")
 				     .to("log:HttpLogger?level=DEBUG&showBody=true&showHeaders=true&multiline=true")
 
-		            // sending message to HIBC for ELIG
-	            .when(simple("${in.header.messageType} == {{hibc-r15-endpoint}} || ${in.header.messageType} == {{hibc-e45-endpoint}}"))
-	                .log("the HIBC endpoint(${in.header.messageType}) is reached and message will be dispatched to message queue(ELIG).")
-                    .setBody(simple(SampleMessages.e45ResponseMessage))
-	            
-	            // sending message to HIBC for ENROL
-	            .when(simple("${in.header.messageType} == {{hibc-r50-endpoint}}"))
-	                .log("the HIBC endpoint (${in.header.messageType}) is reached and message will be dispatched to message queue(ENROL).")
-                    .setBody(simple(SampleMessages.r50ResponseMessage))
+		        // sending message to HIBC for ELIG
+				.when(isMessageForHIBC)
+	                .log("Processing HIBC messages. Request Queue : {{hibc.request.queue}}, ReplyQ:{{hibc.reply.queue}}")
+	                .to("log:HttpLogger?level=DEBUG&showBody=true&showHeaders=true&multiline=true")
+                    .bean(new PopulateJMSMessageHeader()).id("PopulateJMSMessageHeaderHIBC")
+            		.log("HIBC request message ::: ${body}")
+            		.setHeader("CamelJmsDestinationName", constant(String.format("queue:///%s?targetClient=1", System.getenv("HIBC_REQUEST_QUEUE"))))	           		        	
```

Comment:
"queue:///%s?targetClient=1" Could be a re-usable constant (e.g. QUEUE_URI_FORMAT) given that the same value is used for both HIBC and JMB.

Reply:
Added formattable String constant